library/src/main/java/com/google/android/exoplayer2/drm/DefaultDrmSessionManager.java

```diff
@@ -337,6 +343,12 @@ public void setMode(@Mode int mode, byte[] offlineLicenseKeySetId) {
           schemeInitData = psshData;
         }
       }
+      if (C.CENC_UUID.equals(uuid) && MimeTypes.VIDEO_MP4.equals(schemeMimeType)) {
```

Comment:
You should handle audio/mp4 as well for completeness, I think. I'd also rather there were a capping SDK level after which the workaround is not applied. We shouldn't be applying workarounds for platform issues to future platform releases; we should be making sure the problems are fixed in the platform (which it appears has been done).